Troubleshooting: I can't see my tables (#17418)
metabase/metabase: docs/troubleshooting-guide/cant-see-tables.md

> # I can't see my tables

Reviewer: Should we include a table of contents to give people an overview of the potential causes?

Author: Good point

You have connected Metabase to a database, but:

- you don't see the tables in the Data Model section of the Admin Panel,
- the tables don't appear in the Data Browser,
- the tables don't show up as possible data sources when you create a query using the Notebook Editor, or

> - you can no longer see tables that you used to be able to see.

Reviewer: We should link to resources on data model, data browser, as we shouldn't assume readers are familiar with these, especially if they've just set up Metabase.

Author: :+1: added

If you can see the tables, but some of the rows or columns seem to be missing, please check out the [troubleshooting guide for sandboxing][sandboxing].

## Is it a browser caching issue?

**Root cause:** Sometimes browsers will show an old cached list of tables.

**Steps to take:** Refresh your browser tab and check for your table or tables again.

## Does the database exist?

> **Root cause:** The database does not exist. For example, you may have connected to a test database while doing an evaluation but are now in a production environment.
> 

Reviewer: ```suggestion
**Root cause:** The database doesn't exist. For example, you may have connected to a test database while doing an evaluation but are now in a production environment.

```

**Steps to take:**

1. Go to Admin > Databases.

> 2. Check that the database you are trying to query is listed.

Reviewer: ```suggestion
2. Check that the database you're trying to query is listed.
```

3. Click on the database name and examine the settings.

Exactly what settings you need will depend on your environment. To test that the settings are correct:

1. Try to connect to the database using some other application (e.g., `psql` for PostgreSQL).

> If you are not able to connect, the problem is probably not with Metabase. Please check that the database server is running and that you have the correct host, port, username, password, and other settings.

Reviewer: ```suggestion
If you can't connect to the database with another application, the problem is probably not with Metabase. Please check that the database server is running and that you have the correct host, port, username, password, and other settings.
```

> ## Does the table exist and can the Metabase account access it?

Reviewer: As discussed offline, these are two different issues.

Author: split

Be sure to log in using the same credentials that Metabase uses. A common source of problems is that the Metabase "user" does not have the same privileges as a member of IT staff or a developer, so tables that are visible to the latter using external applications are not visible to Metabase.

## Does the person who cannot see the table have permission to view it?

*Root cause:** Metabase uses a group-based permission model: people belong to groups, and administrators can set permissions so that some groups cannot see all of the tables. (It also allows administrators to control which rows or columns specific people can see---issues with that are covered in the troubleshooting guide for [sandboxing][sandboxing].)

**Steps to take:**

1. Log into Metabase using the ID of the person who cannot see the expected tables.
2. Confirm that the tables are not visible.
3. Log out, then log in using the administrator's credentials.

If the administrator's account can see the tables but an individual person cannot:

1. Go to Admin > Permissions and see if any groups have been denied access to the table.

> 2. If any groups have been denied access, go to Admin > People and look at the "Groups" column for the person who cannot see the expected tables. If they are in a group that does not have access to the table, you may need to move them to another group or change table permissions.

Reviewer: ```suggestion
2. If any groups have been denied access, go to Admin > People and look at the "Groups" column for the person who can't see the expected tables. If they're in a group that doesn't have access to the table, you may need to move them to another group or change table permissions.
```

## Is Metabase's metadata out of sync with the state of the database?

**Root cause:** In order to display available tables and columns in dropdown menus and previews, Metabase runs a query every hour to find out what tables are available and what columns are in each available table, and stores this information in its application database.

> 1. If a table has been added or removed since the last time this "sync" operation ran, Metabase's information about the database will be incorrect.

Reviewer: ```suggestion
1. If a table has been added or removed since the last time this "sync" operation ran, Metabase's information about the database will be outdated.
```

> 2. In some rare cases Metabase may time out while synchronizing with the database. For example, if you are using MongoDB and have very large (hundreds of kilobytes) JSON blobs, the sync operation may not complete in the allowed time.

Reviewer: ```suggestion
2. In some rare cases Metabase may time out while synchronizing with the database. For example, if you're using MongoDB and have very large (hundreds of kilobytes) JSON blobs, the sync operation may not complete in the allowed time.
```
